akka-actor/src/main/scala/akka/event/Logging.scala

```diff
@@ -1116,7 +1116,7 @@ trait DiagnosticLoggingAdapter extends LoggingAdapter {
    * These values can be used in PatternLayout when [[akka.event.slf4j.Slf4jLogger]] is configured.
    * Visit <a href="http://logback.qos.ch/manual/mdc.html">Logback Docs: MDC</a> for more information.
    */
-  def setMDC(jMdc: java.util.Map[String, Any]): Unit = mdc(if (jMdc != null) jMdc.asScala.toMap else emptyMDC)
+  def setMDC(jMdc: java.util.Map[String, Any]): Unit = mdc(if (jMdc != null) jMdc.asScala.foldLeft(Map.empty[String, Any])((map, kv) ⇒ map + kv) else emptyMDC)
```

Comment:
more efficient would probably be to skip the `toScala` and iterate `jMdc` entries and convert them to tuples (or use MapBuilder)